pkg/plugins/plugins.go

```diff
@@ -50,19 +53,41 @@ func RegisterSchedulerPlugin(plugin interface{}) {
 		plugins.reconcilePlugin = t
 		registered = true
 	}
+	if t, ok := plugin.(EventPlugin); ok {
+		log.Logger().Debug("register event plugin",
+			zap.String("type", "EventPlugin"))
+		plugins.eventPlugin = t
+		registered = true
+	}
 	if !registered {
 		log.Logger().Debug("no scheduler plugin implemented, none registered")
 	}
 }
 
 func GetPredicatesPlugin() PredicatesPlugin {
+	plugins.Lock()
+	defer plugins.Unlock()
```

Comment:
At most this should be a read lock not a write lock.
However I do not understand why we need a lock. The plugin is only added on register of a RM (i.e. shim) and never manipulated after that. The retrieval only happens in one function. We did not need it before, we should not need it now

Reply:
Without the lock I got data races warnings. Modified the locks to use `RLock` in the get methods.